test/src/main/java/org/apache/accumulo/test/RecoveryIT.java

```diff
@@ -0,0 +1,226 @@
+    // method so that we can modify the properties for each test run
+  }
+
+  @ParameterizedTest
+  @ValueSource(strings = {"TSERVER", "SSERVER", "COMPACTOR"})
+  public void test(String serverForSorting) throws Exception {
+
+    // Determine whether or not we need to disable log sorting
+    // in the TabletServer. We want to do this when the serverForSorting
+    // parameter is SSERVER or COMPACTOR
+    switch (serverForSorting) {
+      case "TSERVER":
+        disableTabletServerLogSorting = false;
+        break;
+      case "SSERVER":
+      case "COMPACTOR":
+      default:
+        disableTabletServerLogSorting = true;
+    }
+
+    // Start the cluster
+    super.setupCluster();
+
+    // create a table
+    String tableName = getUniqueNames(1)[0];
+    try (AccumuloClient c = Accumulo.newClient().from(getClientProps()).build()) {
+      NewTableConfiguration ntc = new NewTableConfiguration();
+      Map<String,String> tableProps = new HashMap<>();
+      tableProps.put(Property.TABLE_MAJC_RATIO.getKey(), "100");
+      tableProps.put(Property.TABLE_FILE_MAX.getKey(), "3");
+      tableProps.put(TableLoadBalancer.TABLE_ASSIGNMENT_GROUP_PROPERTY, RESOURCE_GROUP);
+      ntc.setProperties(tableProps);
+      ntc.withInitialTabletAvailability(TabletAvailability.ONDEMAND);
+      c.tableOperations().create(tableName, ntc);
+
+      TableId tid = TableId.of(c.tableOperations().tableIdMap().get(tableName));
+
+      // create 3 flush files
+      try (BatchWriter bw = c.createBatchWriter(tableName)) {
+        Mutation m = new Mutation("a");
+        m.put("b", "c", new Value("v"));
+        for (int i = 0; i < 3; i++) {
+          bw.addMutation(m);
+          bw.flush();
+          c.tableOperations().flush(tableName, null, null, true);
+        }
+        // create an unsaved mutation
+        bw.addMutation(m);
+      }
+
+      // Confirm that there are no walog entries for this table
+      assertEquals(0, countWaLogEntries(c, tid));
+
+      MiniAccumuloClusterImpl mini = (MiniAccumuloClusterImpl) cluster;
+      MiniAccumuloClusterControl control = (MiniAccumuloClusterControl) cluster.getClusterControl();
+
+      // Stop any running Compactors and ScanServers
+      control.stopAllServers(ServerType.COMPACTOR);
+      control.stopAllServers(ServerType.SCAN_SERVER);
+
+      // Kill the TabletServer in resource group that is hosting the table
+      List<Process> procs = control.getTabletServers(RESOURCE_GROUP);
+      assertEquals(1, procs.size());
+      procs.get(0).destroyForcibly().waitFor();
+      control.getTabletServers(RESOURCE_GROUP).clear();
+
+      // The TabletGroupWatcher in the Manager will notice that the TabletServer is dead
+      // and will assign the TabletServer's walog to all of the tablets that were assigned
+      // to that server. Confirm that walog entries exist for this tablet
+      Wait.waitFor(() -> countWaLogEntries(c, tid) > 0, 60_000);
```

Comment:
For the TSERVER test case logs could be set on the manager and recovered by the tserver w/o this code seeing it.

Reply:
Wonder if the following would work

```suggestion
    if(!serverForSortin.equals("TSERVER")){
      Wait.waitFor(() -> countWaLogEntries(c, tid) > 0, 60_000);
    }
```